### DISCUSSION TITLE
Deprecate (and no longer use) IRepositoryIndexer / IRepositoryDiscoverer

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-m2e/m2e-core#1031] [General]
On[ my journey about rework `IMavenProjectFacade`](https://github.com/eclipse-m2e/m2e-core/issues/1030) I stumbled over some usages of methods requiring a project because of notification of `IRepositoryIndexer` this seems a left-over that is effectively nowhere used in m2e. The indexers are notified about added/removed repositories but nothing else happens.

My idea is therefor to deprecate the `IRepositoryIndexer` interface and remove the remaining usages. The same also seem to apply to `IRepositoryDiscoverer` ... but probably someone knows more about the history and what this is/was useful for?

**Comment by mickaelistria** [comment]
I guess it was used back then when using Maven index to detect artifacts. I don't think it will be missed if you remove them.
